File: backend/app/utils/html_utils.py
"""Pulizia testo da tag HTML e entities (titoli/summary news, ecc.)."""
import re
from html import unescape


def clean_html(text: str | None) -> str:
    """
    Pulizia aggressiva: rimuove tutti i tag HTML, decodifica entities, normalizza spazi.
    Usare su title, summary, subtitle da feed RSS o scraping.
    """
    if not text or not isinstance(text, str):
        return ""
    # Rimuovi tutti i tag HTML (inclusi <p>, <meta>, <br>, <a>, etc.)
    s = re.sub(r"<[^>]+>", "", text)
    # Decodifica HTML entities
    s = unescape(s)
    # Rimuovi spazi multipli e newline
    s = re.sub(r"\s+", " ", s).strip()
    return s

File: frontend_mobile/lib/utils/html_utils.dart
/// Pulizia testo da tag HTML e entities (titoli/summary news).
/// Usare prima di mostrare title/summary/subtitle da API.

String cleanHtml(String text) {
  if (text.isEmpty) return text;
  // Rimuovi tutti i tag HTML
  String s = text.replaceAll(RegExp(r'<[^>]*>'), '');
  // Decodifica entities comuni
  s = s
      .replaceAll('&amp;', '&')
      .replaceAll('&lt;', '<')
      .replaceAll('&gt;', '>')
      .replaceAll('&quot;', '"')
      .replaceAll('&#39;', "'")
      .replaceAll('&rsquo;', "'")
      .replaceAll('&lsquo;', "'")
      .replaceAll('&rdquo;', '"')
      .replaceAll('&ldquo;', '"')
      .replaceAll('&nbsp;', ' ')
      .replaceAll('&egrave;', 'è')
      .replaceAll('&eacute;', 'é')
      .replaceAll('&agrave;', 'à')
      .replaceAll('&ograve;', 'ò')
      .replaceAll('&ugrave;', 'ù')
      .replaceAll('&igrave;', 'ì')
      .replaceAll('&#8217;', "'")
      .replaceAll('&#8216;', "'");
  // Rimuovi spazi multipli
  s = s.replaceAll(RegExp(r'\s+'), ' ').trim();
  return s;
}
